misc_docs/syntax/decorator_deriving.mdx

```diff
@@ -0,0 +1,202 @@
+---
+id: "deriving-decorator"
+keywords: ["deriving", "decorator", "record", "variant"]
+name: "@deriving"
+summary: "This is the `@deriving` decorator."
+category: "decorators"
+---
+
+The `@deriving(accessors)` decorator can be used to either:
```

Comment:
These are outdated examples and docs I believe..
Did you see https://rescript-lang.org/docs/manual/latest/generate-converters-accessors#convert-record-type-to-abstract-record?

The syntax discovery widget should be straight to the point, just point to related docs in a `References` section with a bullet list of links